src/bindings/python/src/pyopenvino/core/tensor.cpp

```diff
@@ -133,6 +133,14 @@ void regclass_Tensor(py::module m) {
             py::arg("type"),
             py::arg("shape"));
 
+    cls.def(py::init<ov::Output<ov::Node>>(), py::arg("node_port"));
+
+    cls.def(py::init([](ov::Output<ov::Node>& port, py::array& array) {
+                return ov::Tensor(port, const_cast<void*>(array.data(0)));
```

Comment:
I have some concerns about this. As previous overload is OK, just creates empty Tensor with given element type and shape. This one might introduce issues with memory.

What happens if passed array is of shape `[2, 2]` let's say `[[1, 2], [3, 4]]` and port has a shape `[2]`? Is result just `[1, 2]`?
So what in the case of port of shape `[5]`? Is the result `[1, 2, 3, 4, 0.000412345154151]`?

What happens if port has mismatched type with array? Let's assume port is `ov.Type.i32` and array is `np.int8`. Is it going to reinterpret data as `int32`? Or is it going to "pack" data into `int32`?

All of this needs to be clarified, tested and documented.

As for shapes there can be a warning that shapes does not match and data will be cropped/extended. (personally I would like a warning about cropping and error about extended as it goes out-of-bounds)

Also warnings about mismatching types would be great. Maybe a combination of both when overall byte size is not matching?

CC @p-wysocki what do you think?

Reply:
added warnings for mismatched elements and shape.